### PR TITLE
Fix: Avoid writing connection string to a same secret

### LINKED ISSUE #152: Avoid writing connection string to a same secret
If two Configurations write a connection string of a cloud resource to the same secret, the data will mix up.

```
apiVersion: terraform.core.oam.dev/v1beta1
kind: Configuration
metadata:
  name: alibaba-rds-1
spec:
  writeConnectionSecretToRef:
    name: rds-conn
    namespace: default
```

```
apiVersion: terraform.core.oam.dev/v1beta1
kind: Configuration
metadata:
  name: alibaba-rds-2
spec:
  writeConnectionSecretToRef:
    name: rds-conn
    namespace: default
```

For example, the data of rds-conn will get mixed up.

### PR DESCRIPTION
**Pull request by loheagn**
This pr fixes #152 .

Signed-off-by: loheagn <loheagn@icloud.com>

### CODE REVIEW

**Comment by zzxwill** [comment]
Please add UTs for this feature. Thanks.

**Comment by loheagn** [comment]
> Please add UTs for this feature. Thanks.

Sorry for the late reply. I've been a little busy lately and may not be able to start this work until this weekend.

**Comment by msftgits** [comment]
[CLA assistant check](https://cla.opensource.microsoft.com/oam-dev/terraform-controller?pullRequest=250) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/oam-dev/terraform-controller?pullRequest=250) before we can accept your contribution.<br/><br/>:x: loheagn [sign now](https://cla.opensource.microsoft.com/oam-dev/terraform-controller?pullRequest=250)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/oam-dev/terraform-controller?pullRequest=250) it.</sub>

File: controllers/configuration_controller.go [hunk 1]
```diff
@@ -743,24 +743,32 @@ func (meta *TFConfigurationMeta) getTFOutputs(ctx context.Context, k8sClient cli
...
 	if err := k8sClient.Get(ctx, client.ObjectKey{Name: name, Namespace: ns}, &gotSecret); err != nil {
 		if kerrors.IsNotFound(err) {
 			var secret = v1.Secret{
 				ObjectMeta: metav1.ObjectMeta{
 					Name:      name,
 					Namespace: ns,
 					Labels: map[string]string{
-						"created-by": "terraform-controller",
+						"created-by":                 "terraform-controller",
+						"terraform-controller-owner": configurationName,
```
[the remark this answers is not in the document]
Comment (loheagn) [reply controllers/configuration_controller.go:797]: Yes. I will fix it.

[the remark this answers is not in the document]
Comment (loheagn) [reply controllers/configuration_controller.go:797]: Yes. If so, does both namespace and name need to be checked when try to update a secret?

[the remark this answers is not in the document]
Comment (loheagn) [reply controllers/configuration_controller.go:797]: I don't think it is necessary. If the name of the configuration which want to update a secret is different than the secret's owner, the update operation SHOULD be prohibited, regardless of whether they belong to the same or different namespaces.

[the remark this answers is not in the document]
Comment (loheagn) [reply controllers/configuration_controller.go:797]: Sorry, I am wrong. It's necessary to check the namespace when the configurations have the same name……

File: controllers/configuration_controller.go [hunk 2]
```diff
@@ -785,24 +785,43 @@ func (meta *TFConfigurationMeta) getTFOutputs(ctx context.Context, k8sClient cli
...
 			}
 		}
 	} else {
+		// check the owner of this secret
+		labels := gotSecret.ObjectMeta.Labels
+		ownerName := labels["owned-by"]
+		ownerNamespace := labels["owned-namespace"]
+		if configurationName != ownerName || configuration.Namespace != ownerNamespace {
+			errMsg := fmt.Sprintf(
+				"configuration(%s-%s) cannot update secret(%s) which owner is configuration(%s-%s)",
```
Comment (zzxwill) [inline controllers/configuration_controller.go:818]:
```suggestion
				"configuration(%s-%s) cannot update secret(%s) whose owner is configuration(%s-%s)",
```

**Review by zzxwill — APPROVED** [review]
Good job!

File: controllers/configuration_controller.go (hunk 2 above)
Comment (loheagn) [reply controllers/configuration_controller.go]: Fixed :)

**Review by Nicola115 — APPROVED** [review]

**Pull request merged**